### ISSUE TITLE
Allow surfers to save "favourite" pages

### ISSUE THREAD

**Issue by alcinnz** [alcinnz/Odysseus#78]
Another important, but reportedly underutilized, browser feature is bookmarks/favourite pages.

In terms of UI:

* Favourites should be organized by a set of "tags", and not into a folder hierarchy. This'll lighten the [organizational burden](https://zachholman.com/posts/shit-work/) to the point where the favourited pages could almost help out.
* A new toolbar item should be added after the addressbar for adding and removing the current page from favourites. This should bring up a popover form that helps you organize the page with strong defaults and a touch of intelligence.
* The main way for accessing bookmarks should be by typing known tags into the addressbar, thereby performing a search. A [Toast](https://valadoc.org/granite/Granite.Widgets.Toast.html) should be used to educate surfers to do this upon bookmarking a page.
* The bookmarks page should also be accessible via the Application Menu, and should list the available tags in a sidebar.
* Support for [SKOS](https://www.w3.org/2004/02/skos/) should be implemented to allow pages to do a great job of organizing themselves into your favourites, and to help open up the possibility to send subsections of your bookmarks to someone else.
* Bookmarks links should be depicted as a captioned screenshot, with a configurable tooltip.

**Comment by alcinnz** [comment]
To split this into subtasks:

- [ ] Add favourites to the database.
- [ ] Render a tokenized text input, which looks just like a text input but contains rounded-box tokens (in part just to distinguish normal URIs from bookmarks search).
- [ ] Use the tokenized text input for the addressbar.
- [ ] Add a toolbar button to indicate whether a page is favourited and to edit it's details. Use the tokenized text input as the main control of the edit form.
- [ ] Create odysseus:favourites, to view and search these links by tag.
- [ ] Integrate odysseus:favourites into the newly tokenized addressbar.
- [ ] Link to odysseus:favourites from the Application Menu.

**Comment by alcinnz** [comment]
Depends on #75, for the infrastructure to store and present these links.

**Comment by alcinnz** [comment]
Why's this closed?
